app/jobs/regular/export_csv_file.rb

```diff
@@ -29,15 +30,22 @@ def execute(args)
       @extra = HashWithIndifferentAccess.new(args[:args]) if args[:args]
       @current_user = User.find_by(id: args[:user_id])
 
-      export_method = :"#{@entity}_export"
-      raise Discourse::InvalidParameters.new(:entity) unless respond_to?(export_method)
+      entities = [{ name: @entity }]
+      entities << { name: "user_archive_profile" } if @entity === "user_archive"
 
-      file_name_prefix = if @entity == "user_archive"
-        "#{@entity.split('_').join('-')}-#{@current_user.username}-#{Time.now.strftime("%y%m%d-%H%M%S")}"
-      elsif @entity == "report" && @extra[:name].present?
-        "#{@extra[:name].split('_').join('-')}-#{Time.now.strftime("%y%m%d-%H%M%S")}"
-      else
-        "#{@entity.split('_').join('-')}-#{Time.now.strftime("%y%m%d-%H%M%S")}"
+      entities.each do |entity|
+        entity[:method] = :"#{entity[:name]}_export"
+        raise Discourse::InvalidParameters.new(:entity) unless respond_to?(entity[:method])
+
+        @timestamp ||= Time.now.strftime("%y%m%d-%H%M%S")
+        entity[:filename] =
+          if entity[:name] == "user_archive" || entity[:name] === "user_archive_profile"
+            "#{entity[:name].split('_').join('-')}-#{@current_user.username}-#{@timestamp}"
```

Comment:
This would read a lot easier if you put `#{entity[:name].split('_').join('-')}` into a variable

Reply:
That's right. I just moved the code and did not realize how weird that line is (why not `gsub` in the first place?). I am using `dasherize` now which is more succinct and hopefully does a better job at formatting.